### DISCUSSION TITLE
Difficulty interpreting JSON result

### DISCUSSION THREAD

**Discussion by NathanM0153** [ScrappyCocco/HowLongToBeat-PythonAPI#21] [Q&A] [answered]
Hello,

I'm having trouble doing anything with the result of the .search() method. Below is my function using it:

def findGame(name): 
    results = HowLongToBeat().search(name, similarity_case_sensitive=False)
    if results is not None and len(results) > 0:
        best_element = max(results, key=lambda element: element.similarity)
        return best_element
    else:
        print("Not found")
        return

edit: this doesn't seem to be indenting correctly in this interface, but it obviously is indented in code.

And the output:

findGame("Portal")
<howlongtobeatpy.HowLongToBeatEntry.HowLongToBeatEntry object at 0x00000276CC7E2C50>

What exactly does this do, if I can't just return it? I can't find any info on how to interpret this in the JSONResultParser file, it doesn't seem like I can call the parse_web_result method directly, and looking through the main file it seems like should automatically decode this. The only SO posts I can find about this are about user-created objects that can't be displayed naturally, but .search() should return a list from what I see, not a custom object that has no good way to be displayed.

**Comment by ScrappyCocco** [comment, marked as answer]
Hello @NathanM0153 and thanks for writing here 
I think you are missing the fact that the search return an object of type [HowLongToBeatEntry.py](https://github.com/ScrappyCocco/HowLongToBeat-PythonAPI/blob/master/howlongtobeatpy/howlongtobeatpy/HowLongToBeatEntry.py), looking at the [test cases](https://github.com/ScrappyCocco/HowLongToBeat-PythonAPI/blob/master/howlongtobeatpy/tests/test_normal_request.py) should help you understand how to use and read the returned object.

Here's the code you tried to write that correctly display the results accessing the struct
```python
from howlongtobeatpy import HowLongToBeat

def findGame(name):
    results = HowLongToBeat().search(name, similarity_case_sensitive=False)
    if results is not None and len(results) > 0:
        best_element = max(results, key=lambda element: element.similarity)
        return best_element
    else:
        print("Not found")
        return

game = findGame("Portal")
print(str(game.game_name))
print(str(game.game_web_link))
print(str(game.profile_platforms))
```
This code correctly prints this
```
Portal
https://howlongtobeat.com/game/7230
['Mac', 'Nintendo Switch', 'PC', 'PlayStation 3', 'Xbox 360']
```
Please let me know if you understood or if you need additional help

**Reply by NathanM0153** [reply]
Thanks for replying! That makes sense.

**Reply by rabblerowser** [reply]
Looking at this a few years later, this was how I found out how to work with the hltbobject. Not sure if there's a way to get this added into the documentation under a "object manipulation" portion

**Reply by ScrappyCocco** [reply]
@rabblerowser what do you thing is missing from the actual documentation? It looks pretty clear to me

**Reply by rabblerowser** [reply]
@ScrappyCocco the details about how to interact with the hltb Object should be part of the documentation, but it is not in there and only in the code for the object. If that's your intention, then maybe listing out a separate page for the object interactions? It just seems weird to me to have documentation that points to code as its explanation tool.
